vassal-app/src/main/java/VASSAL/configure/ConfigureTree.java

```diff
@@ -2513,119 +2880,206 @@ void startNewTrait() {
+      final Configurable c = (Configurable) node.getUserObject();
+      final String item = getConfigureName(c.getClass());
+      final String name = StringUtils.defaultString(c.getConfigureName());
+      final String matchString = getString("Editor.search_matches", nodeListIndex) + "<b>" + noHTML(name + " [" + item + "]") + "</b>: ";
+      final TargetProgress progress = new TargetProgress();
+      showConfigurableHitList(node, regexPattern, matchString, progress);
+    }
 
-    private void showConfigurableHitList(DefaultMutableTreeNode node, String searchString) {
+    /**
+     * Called from showHitList to generate detailed output for Configurables.
+     * @param node - any node of our module tree
+     * @param regexPattern - our search string
+     * @param progress - tracks whether component has matched already (ensures only first-time hit generates header output)
+     */
+    private void showConfigurableHitList(DefaultMutableTreeNode node, Pattern regexPattern, String matchString, TargetProgress progress) {
       final Configurable c = (Configurable) node.getUserObject();
-      if (!(c instanceof SearchTarget)) {
-        return;
+      final String item = getConfigureName(c.getClass());
+      final boolean showName = (searchParameters.isMatchNames() || !searchParameters.isMatchAdvanced());  // name is default (i.e. unless filtered out)
+
+      // Help ?
+      try {
+        // FIXME: Name check is internal - comparison should ideally be on raw data, not translated
+        if (getConfigureName((Configurable) node.getParent()).equals(getString("Editor.Documentation.component_type"))) {
+          // Help - special processing to include in full search despite not being a SearchTarget
+          // Menu is also the Name, so only display here if not already reported as Name
+          if (!showName && (searchParameters.isMatchFull() || searchParameters.isMatchMenus())) {
+            stringListHits(true, Arrays.asList(c.getConfigureName()), regexPattern, matchString, item, "", "UI Text", progress); //NON-NLS
+          }
+          if (searchParameters.isMatchFull() || searchParameters.isMatchExpressions()) {
+            // content refs will be categorised as Expressions
+            final String startPage = c.getAttributeValueString("startingPage");
+            final String pdfFile = c.getAttributeValueString("pdfFile");
+            final String textFile = c.getAttributeValueString("textFile");
+            stringListHits(true, Collections.singletonList(startPage), regexPattern, matchString, item, "", "Starting page", progress); //NON-NLS
+            stringListHits(true, Collections.singletonList(pdfFile), regexPattern, matchString, item, "", "PDF file", progress); //NON-NLS
+            stringListHits(true, Collections.singletonList(textFile), regexPattern, matchString, item, "", "Text File", progress); //NON-NLS
+          }
+        }
       }
+      catch (Exception NullPointerException) {
```

Comment:
What are the exceptions which can actually be thrown here? NPE should never be caught---the code which can throw it should be fixed.